### PR TITLE
feature: http buffer filter configuration (#6249)

### PR DESCRIPTION
**Pull request by ruzmuh**
Proposal for Issue #6249:
Introduce two optional command line flags, `envoy-http-buffer-max-request-bytes` and `envoy-https-buffer-max-request-bytes`, with default values set to `0`. If the value is non-zero, an HTTP buffer filter will be added to the HTTP filter chain immediately after the `DefaultFilters()` with the `max_request_bytes` parameter. This configuration allows setting the buffer filter for the entire HTTP listener. In the future, it possible to provide ability to override these values on a per-route basis using [bufferperroute](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/buffer/v3/buffer.proto#extensions-filters-http-buffer-v3-bufferperroute).

### CODE REVIEW

File: cmd/contour/serve.go [hunk 1]
```diff
@@ -137,7 +137,9 @@ func registerServe(app *kingpin.Application) (*kingpin.CmdClause, *serveContext)
 	serve.Flag("disable-leader-election", "Disable leader election mechanism.").BoolVar(&ctx.LeaderElection.Disable)
 
 	serve.Flag("envoy-http-access-log", "Envoy HTTP access log.").PlaceHolder("/path/to/file").StringVar(&ctx.httpAccessLog)
+	serve.Flag("envoy-http-buffer-max-request-bytes", "Envoy HTTP buffer max request bytes.").PlaceHolder("10485760").Uint32Var(&ctx.httpBufferMaxRequestBytes)
 	serve.Flag("envoy-https-access-log", "Envoy HTTPS access log.").PlaceHolder("/path/to/file").StringVar(&ctx.httpsAccessLog)
+	serve.Flag("envoy-https-buffer-max-request-bytesg", "Envoy HTTPS buffer max request bytes.").PlaceHolder("10485760").Uint32Var(&ctx.httpsBufferMaxRequestBytes)
```
Comment (lubronzhan) [inline cmd/contour/serve.go:142]:
```suggestion
	serve.Flag("envoy-https-buffer-max-request-bytes", "Envoy HTTPS buffer max request bytes.").PlaceHolder("10485760").Uint32Var(&ctx.httpsBufferMaxRequestBytes)
```

File: cmd/contour/serve.go [hunk 2]
```diff
@@ -137,7 +137,9 @@ func registerServe(app *kingpin.Application) (*kingpin.CmdClause, *serveContext)
 	serve.Flag("disable-leader-election", "Disable leader election mechanism.").BoolVar(&ctx.LeaderElection.Disable)
 
 	serve.Flag("envoy-http-access-log", "Envoy HTTP access log.").PlaceHolder("/path/to/file").StringVar(&ctx.httpAccessLog)
+	serve.Flag("envoy-http-buffer-max-request-bytes", "Envoy HTTP buffer max request bytes.").PlaceHolder("10485760").Uint32Var(&ctx.httpBufferMaxRequestBytes)
```
Comment (sunjayBhatia) [inline cmd/contour/serve.go:140]: i dont think we want to add command line flags for this, rather for now just add configuration options to the configuration CRD and config file schema (https://github.com/projectcontour/contour/blob/main/pkg/config/parameters.go)

is there a reason you want to add a separate limit for http vs. https?

Comment (ruzmuh) [reply cmd/contour/serve.go:140]: > for now just add configuration options to the configuration CRD and config file schema
ok, I'll try to do that. 

> is there a reason you want to add a separate limit for http vs. https?
no, there wasn't any specific reason behind it; it was just a proposal.  If you believe that merging them into a single parameter would be more effective, then let's proceed with that approach.

Comment (ruzmuh) [reply cmd/contour/serve.go]: fixed

**Comment by frankh** [comment]
can this be reopened? It's a pretty important feature

**Comment by bciaraldi** [comment]
I also would like to request that this be reopened.

**Comment by Piccirello** [comment]
@skriss is there any interest in merging this PR? As the issue notes, this helps protect upstream services from slow requests, aka a [slowloris DDoS attack](https://www.cloudflare.com/learning/ddos/ddos-attack-tools/slowloris/). This is already supported upstream in Envoy and just needs to be exposed in Contour.

**Pull request closed without merging**